src/rust/grapl-web-sec/src/authn/user.rs

```diff
@@ -0,0 +1,82 @@
+use std::pin::Pin;
+
+use actix_web::{
+    dev::Payload,
+    FromRequest,
+    HttpRequest,
+};
+use futures::Future;
+use grapl_config::env_helpers::FromEnv;
+use rusoto_dynamodb::DynamoDbClient;
+
+use crate::{
+    authn::GraplRole,
+    error::WebSecError,
+};
+
+const SESSION_TOKEN: &'static str = "SESSION_TOKEN";
+
+/// Represents an Authenticated User
+#[derive(Clone)]
+pub struct AuthenticatedUser {
+    identity: String,
+    role: GraplRole,
+}
+
+impl AuthenticatedUser {
+    pub fn get_identity(&self) -> &str {
+        &self.identity
+    }
+
+    pub fn get_role(&self) -> &GraplRole {
+        &self.role
+    }
+}
+
+impl AuthenticatedUser {
+    #[allow(dead_code)]
+    #[cfg(test)]
+    /// Used only for constructing tests; should NEVER be used in production code
```

Comment:
It seems like this code should be reachable from production if that's the case, but also I'm a little confused as to this in the first place: it looks more like a `new` function and I don't see why it should be test specific?